tajo-docs/src/main/sphinx/configuration/tajo-site-xml.rst

```diff
@@ -2,23 +2,455 @@
+
+=========================
+Group by Query Settings
+=========================
+
+""""""""""""""""""""""""""""""""""""""""""""
+`tajo.dist-query.groupby.multi-level-aggr`
+""""""""""""""""""""""""""""""""""""""""""""
+
+A flag to enable the multi-level algorithm for distinct aggregation. If this value is set, 3-phase aggregation algorithm is used.
+Otherwise, 2-phase aggregation algorithm is used.
+
+  * Property value: Boolean
+  * Default value: true
+  * Example
+
+.. code-block:: xml
+
+  <property>
+    <name>tajo.dist-query.groupby.multi-level-aggr</name>
+    <value>true</value>
+  </property>
+
+""""""""""""""""""""""""""""""""""""""""""""""
+`tajo.dist-query.groupby.partition-volume-mb`
+""""""""""""""""""""""""""""""""""""""""""""""
+
+The aggregation is executed in two stages. When an aggregation query is executed,
+this value indicates the output size of each task at the first stage, which determines the number of partitions to be shuffled between two stages.
+
+  * Property value: Integer
+  * Unit: MB
+  * Default value: 256
+  * Example
+
+.. code-block:: xml
+
+  <property>
+    <name>tajo.dist-query.groupby.partition-volume-mb</name>
+    <value>256</value>
+  </property>
+
+""""""""""""""""""""""""""""""""""""""""""""""
+`tajo.dist-query.groupby.task-volume-mb`
+""""""""""""""""""""""""""""""""""""""""""""""
+
+The aggregation operation is executed in two stages. When an aggregation query is executed, this value indicates the amount of input data processed by each task at the second stage.
+As a result, it determines the degree of the parallel processing of the aggregation query.
+
+  * Property value: Integer
+  * Unit: MB
+  * Default value: 64
+  * Example
+
+.. code-block:: xml
+
+  <property>
+    <name>tajo.dist-query.groupby.partition-volume-mb</name>
+    <value>64</value>
+  </property>
+
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+`tajo.executor.groupby.in-memory-hash-threshold-mb`
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+This value provides the criterion to decide the algorithm to perform an aggregation in a task.
+If the input data is smaller than this value, the aggregation is performed with the in-memory hash aggregation.
+Otherwise, the sort-based aggregation is used.
+
+  * Property value: Integer
+  * Unit: MB
+  * Default value: 64
+  * Example
+
+.. code-block:: xml
+
+  <property>
+    <name>tajo.executor.groupby.in-memory-hash-threshold-mb</name>
+    <value>64</value>
+  </property>
+
+.. warning::
+  This value is the size of the input stored on file systems. So, when the input data is loaded into JVM heap,
+  its actual size is usually much larger than the configured value, which means that too large threshold can cause unexpected OutOfMemory errors.
+  This value should be tuned carefully.
+
+""""""""""""""""""""""""""""""""""""""""""
+`tajo.executor.aggregate.hash-table.size`
+""""""""""""""""""""""""""""""""""""""""""
+
+The initial size of list for in-memory sort.
```

Comment:
Description explains for list size, but property name looks that it means hash table size.